ofrak_core/test_ofrak/components/test_string.py

```diff
@@ -1,16 +1,58 @@
+import os
 import pytest
+import subprocess
+
+from typing import List
 
 from ofrak import OFRAKContext
 from ofrak.component.modifier import ModifierError
 from ofrak.core.binary import GenericBinary
 from ofrak.resource import Resource
+from ofrak.service.resource_service_i import ResourceFilter
 from ofrak.core.strings import (
+    AsciiString,
     StringPatchingConfig,
     StringPatchingModifier,
     StringFindReplaceConfig,
     StringFindReplaceModifier,
 )
 
+STRING_TEST_C_SOURCE = r"""
+#include <stdio.h>
+
+extern int longString(void);
+extern int shortString(void);
+
+// Generate bytes that look like a long ascii string (21 bytes)
+__asm__(".global longString\n\t"
+    ".type longString, @function\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "push %r15\n\t"
+    "and 0, %r15\n\t"
+);
+
+// Generate bytes that look like a short ascii string (7 bytes)
+__asm__(".global shortString\n\t"
+    ".type shortString, @function\n\t"
+    "push %r15\n\t"
+    "and 0, %r15\n\t"
+);
```

Comment:
The implication is that these shouldn't match as strings, but will, right?

Might be worth clarifying that in the comments?

Reply:
The only thing stopping these being matched as a string should be the min length requirement (8) for strings intermixed with code.